external-crates/move/tools/move-package/src/resolution/dependency_graph.rs

```diff
@@ -667,60 +689,46 @@ impl DependencyGraph {
-                                        &other_g,
-                                        *dep_name,
-                                        *is_other_external
-                                    )?,
-                                    other_pkg_deps
-                                ),
-                                format_deps(
-                                    dep_path_from_root(root_package, &g, *dep_name, *is_external)?,
-                                    pkg_deps
-                                ),
-                            )
-                        }
-                    }
-                }
-            }
+        Ok(())
+    }
 
-            if internally_resolved {
-                // insert edges from the directly dependent package to its neighbors for
-                // internally resolved sub-graphs - due to how external graphs are constructed,
-                // edges between directly dependent packages and their neighbors are already in
-                // the sub-graph and would have been inserted in the first loop in this function
-                for (_, to_pkg_name, sub_dep) in g.package_graph.edges(*dep_name) {
-                    self.package_graph
-                        .add_edge(*dep_name, to_pkg_name, sub_dep.clone());
-                }
-            }
+    /// The merge algorithm relies on the combined graph to be pre-populated with direct
+    /// dependencies for internally resolved packages (in terms of both entries in the package table
+    /// and nodes/edges in the package graph). This means that if a conflict is detected between
+    /// pre-populated combined graph and one of the sub-graphs, the pre-populated graph does not
+    /// (yet) contains target packages for edges outgoing from direct dependencies, and these are
+    /// needed to verify that direct dependencies are the same. Fortunately, these packages are
+    /// available in a package tables of a respective (dependent) subgraph. This function return the
+    /// right package table, depending on whether conflict was detected between pre-populated
+    /// combined graph and another sub-graph or between two separate sub-graphs. If we tried to use
+    /// combined graphs's package table "as is" we would get an error in all cases similar to the on
+    /// in the direct_and_indirect_dep test where A is a direct dependency of Root (as C would be
+    /// missing from the combined graph's table):
+    ///
+    ///                 +----+
+    ///           +---->| B  |----+
+    ///           |     +----+    |
+    ///           |               |
+    /// +----+    |               +-->+----+     +----+
+    /// |Root|----+------------------>| A  |---->| C  |
+    /// +----+                        +----+     +----+
+    ///
+    fn pkg_table_for_deps_compare<'a>(
+        &self,
+        dep_name: Symbol,
+        g: &'a DependencyGraph,
+        dep_graphs: &'a BTreeMap<PM::PackageName, (DependencyGraph, DependencyMode, bool, bool)>,
```

Comment:
nit: The value type is verging on needing a `struct` for clarity, for me.

Reply:
Good idea!